R/util_data_summaries.R

```diff
@@ -126,3 +155,509 @@ ds.summarize_by_column <- function(
+        msl$mod_se <- coefs[3,2]
+        msl$mod_stat <- coefs[3,3]
+        msl$mod_p <- coefs[3,4]
+        
+        int_row <- nrow(coefs)
+        
+        msl$int_coef <- coefs[int_row,1]
+        msl$int_se <- coefs[int_row,2]
+        msl$int_stat <- coefs[int_row,3]
+        msl$int_p <- coefs[int_row,4]
+
+        if(msl$family %in% "binomial"){
+            msl$int_apa <- "OR=" %+% round(exp(msl$int_coef),2) %+%
+                ", z=" %+% round(msl$int_stat,3) %+%
+                ifelse(msl$int_p < .001,", p<.001",", p=" %+% round(msl$int_p,3))
+            
+        } else if(msl$family %in% "gaussian"){
+            msl$int_apa <- "b=" %+% round(msl$int_coef,2)  %+%
+                ", t(" %+% msl$df %+% ")" %+% "=" %+% 
+                round(msl$int_stat,3) %+%
+                ifelse(msl$int_p < .001, 
+                       ", p<.001", 
+                       ", p=" %+% round(msl$int_p,3)
+                )
+        }
+    } else{
+        msl$int_apa <- ""
+    }
+    
+    return(msl)
+}
+
+#########################################################
+###
+###     General Model Summary Function(s)
+###     Should work on models outside of glm1 spec.
+###
+#########################################################
+
+ds.model_summary_df <- function( 
+    models, # list of models
+    summary_func = ds.summarize_glm1  # extracts ds.glm1 features
+){
+    # returns a data.frame with one row per supplied model
+    # and one column for each list index returned by the
+    # supplied summary_func
+    # example use cases of output:
+    #   sort or filter summary data.frame for specific models
+    #   pass to printing functions to nice summary tables
+    
+    # msls = model summary lists
+    msls <- lapply( models , function(glm_model){
+        as.data.frame(summary_func(glm_model))
+    })
+    
+    # remove empty feature lists
+    # (e.g., may happen if ds.glm1 spec is violated)
+    sanitized_msls <- list()
+    for( i in 1:length(msls) ){
+        if( nrow(msls[[i]]) > 0 ){
+            sanitized_msls[[length(sanitized_msls) + 1]] <- msls[[i]]
+        }
+    }
+    if(length(sanitized_msls) > 1){
+        feature_table <- util.rbind_union(sanitized_msls)
+    } else if(length(sanitized_msls) == 1){
+        feature_table <- sanitized_msls[[1]]
+    } else{
+        feature_table <- data.frame()
+    }
+    return(feature_table)
+}
+
+
+#########################################################
+###
+###     Unit Tests
+###
+#########################################################
+
+ds.helper$unit_test <- function(){
+    
+    assert_expected_output <- function( func_name , input, expected_output ){
+        # warn if evaluated func_name(input) != expected_output
+        # errors messages are not compared, but errors can be expected
+        
+        # func_name passed as string rathern than func itself
+        # so the func_name can be included warnings
+        func <- eval(parse(text=func_name))
```

Comment:
I tried this, and it doesn't work. I think `get` only retrieves complete objects in the workspace, not on their children. See below.

This works:
```r
get("ds.helper")
```

But this:
```r
get("ds.helper$variable_type")
```
Returns
```
Error in get("ds.helper$variable_type") : 
  object 'ds.helper$variable_type' not found
```

Anyway, I'm leaving this as is.